### PR TITLE
replace deprecated types

### PR DESCRIPTION
**Pull request by chirag-parmar**
### Changelog

- rename `EthAddress` to `Address` to 'possibly' reduce imports.
- rename deprecated types `Hash256`, `BloomFilter`, `BlockHash`, `Blob`, `BlockHeader`, `VersionedHashes` and more.
- rename `EthBlock` to `Block` to 'possibly' reduce imports.
- remove usage `w3Addr`, `w3Hash`, `ethHash`, `ethAddr`  and other conversions wherever possible.
- use specific imports of types wherever possible

### CODE REVIEW

**Comment by arnetheduck** [comment]
Rather than a straight search and replace, the plan here is to also update imports at the same time so as to move away from importing eth/common and instead import more specific types - in particular, most places that use `Hash256` actually only need to import `eth/common/hashes`.

The other part of this change is to remove unnecessary conversions, such as `w3hash` and friends since the types in nim-eth and nim-web3 now match. The warnings are a useful marker for code that hasn't yet been reviewed.

File: nimbus/common/chain_config.nim [hunk 1]
```diff
@@ -11,7 +11,8 @@
 
 import
   std/[tables, strutils, times, macros],
-  eth/[common, rlp, p2p], eth/common/eth_types_json_serialization,
+  eth/[rlp, p2p], eth/common/eth_types_json_serialization,
+  eth/common/[base, hashes, addresses],
```
Comment (jangko) [inline nimbus/common/chain_config.nim:15]: Looks like you need to import `eth/common/eth_types_rlp`  too to solve the failure in CI.

Comment (chirag-parmar) [reply nimbus/common/chain_config.nim:15]: Thanks @jangko

**Comment by chirag-parmar** [comment]
@kdeme @arnetheduck this is ready for review.

File: nimbus/beacon/web3_eth_conv.nim [hunk 2]
```diff
@@ -25,14 +25,14 @@ export
 
 type
   Web3FixedBytes*[N: static int] = web3Types.FixedBytes[N]
```
Comment (jangko) [inline nimbus/beacon/web3_eth_conv.nim:27]: No need to keep `Web3FixedBytes`, now it is the same type with nim-eth `FixedBytes`.

Ditto for `Web3Hash`, `Web3Address`, `Web3Bloom`, and `Web3PrevRandao`.

Comment (chirag-parmar) [reply nimbus/beacon/web3_eth_conv.nim:27]: Had removed it earlier but reverted the changes. I'm not sure if ditto types have any use in Nim but I reverted the changes because they didn't cause any deprecation warnings. Will remove them again.

**Comment by advaita-saha** [comment]
> #2733 might address the Kurtosis CI failure

Yes I rebase with master will solve the kurtosis issue

**Comment by jangko** [comment]
tips: apart from fluffy, this kind of types and identifiers job usually need to build/compile:
- nimbus/nimbus
- tests/all_tests
- tests/test_tools_build
- nrpc/nrpc

if these four can be built using `nim c -c ...` usually there will be no problem in the CI(excluding logical bugs).

File: hive_integration/nodocker/engine/engine/invalid_payload.nim [hunk 3]
```diff
@@ -166,7 +166,7 @@ method execute(cs: InvalidPayloadTestCase, env: TestEnv): bool =
 
       var attr = env.clMock.latestPayloadAttributes
       attr.timestamp = w3Qty(shadow.alteredPayload.timestamp, 1)
-      attr.prevRandao = w3Hash()
+      attr.prevRandao = Hash32()
```
Comment (jangko) [inline hive_integration/nodocker/engine/engine/invalid_payload.nim:169]: same as above

File: hive_integration/nodocker/engine/engine/payload_execution.nim [hunk 4]
```diff
@@ -349,7 +349,7 @@ method execute(cs: NewPayloadOnSyncingClientTest, env: TestEnv): bool =
 
       # Send the forkchoiceUpdated with a reference to the valid payload on the SYNCING client.
       var
-        random                = w3Hash()
+        random                = Hash32()
```
Comment (jangko) [inline hive_integration/nodocker/engine/engine/payload_execution.nim:352]: same as above

File: hive_integration/nodocker/rpc/client.nim [hunk 5]
```diff
@@ -66,7 +66,7 @@ proc txReceipt*(client: RpcClient, txHash: eth_types.Hash32): Future[Option[Rece
     receiptType: LegacyReceipt,
     isHash     : rc.root.isSome,
     status     : rc.status.isSome,
-    hash       : rc.root.get(w3Hash()),
+    hash       : rc.root.get(),
```
Comment (jangko) [inline hive_integration/nodocker/rpc/client.nim:69]: should be `hash       : rc.root.get(default(Hash32))`

File: hive_integration/nodocker/engine/withdrawals/wd_base_spec.nim [hunk 6]
```diff
@@ -160,12 +160,12 @@ proc verifyContractsStorage(ws: WDBaseSpec, env: TestEnv): Result[void, string]
 
   if latestPayloadNumber >= ws.forkHeight.uint64:
     # Shanghai
-    r.expectStorageEqual(WARM_COINBASE_ADDRESS, 100.u256.w3FixedBytes)    # WARM_STORAGE_READ_COST
-    p.expectStorageEqual(PUSH0_ADDRESS, latestPayloadNumber.u256.w3FixedBytes) # tx succeeded
+    r.expectStorageEqual(WARM_COINBASE_ADDRESS, FixedBytes[32](100.u256.toBytesBE))    # WARM_STORAGE_READ_COST
```
Comment (jangko) [inline hive_integration/nodocker/engine/withdrawals/wd_base_spec.nim:163]: nim-eth have something like `100.u256.to(Bytes32)` for this kind of conversion.

Comment (chirag-parmar) [reply hive_integration/nodocker/engine/withdrawals/wd_base_spec.nim:163]: very handy!! will change this. Thanks for the suggestion.

File: hive_integration/nodocker/rpc/client.nim (hunk 5 above)
Comment (chirag-parmar) [reply hive_integration/nodocker/rpc/client.nim:69]: Hmm don't know how I missed it. Thanks

File: hive_integration/nodocker/engine/types.nim [hunk 7]
```diff
@@ -103,11 +103,11 @@ template testCond*(expr, body: untyped) =
     body
     return false
 
-proc `==`*(a: Opt[BlockHash], b: Opt[common.Hash256]): bool =
+proc `==`*(a: Opt[Hash32], b: Opt[Hash32]): bool =
```
Comment (arnetheduck) [inline hive_integration/nodocker/engine/types.nim:106]: this function can be removed entirely

File: nimbus/beacon/beacon_engine.nim [hunk 8]
```diff
@@ -85,8 +84,8 @@ template wrapException(body: untyped): auto =
 
 # setInvalidAncestor is a callback for the downloader to notify us if a bad block
 # is encountered during the async sync.
-func setInvalidAncestor(ben: BeaconEngineRef,
-                         invalid, origin: common.BlockHeader) =
+proc setInvalidAncestor(ben: BeaconEngineRef,
```
Comment (arnetheduck) [inline nimbus/beacon/beacon_engine.nim:87]: we tend to prefer `func` when that works, why the change here?

File: hive_integration/nodocker/engine/types.nim (hunk 7 above)
Comment (chirag-parmar) [reply hive_integration/nodocker/engine/types.nim:106]: makes sense. missed it.

File: nimbus/beacon/beacon_engine.nim (hunk 8 above)
Comment (chirag-parmar) [reply nimbus/beacon/beacon_engine.nim:87]: good catch. I think I missed the change [here](https://github.com/status-im/nimbus-eth1/commit/266b72698deeab5c7742f144af38dfe2738c466c) while rebasing. Should I open another PR?

Comment (arnetheduck) [reply nimbus/beacon/beacon_engine.nim:87]: yes
